### PR TITLE
Bundle lib 0.2

### PR DESCRIPTION
**Pull request by shawn-hurley**
#### Describe what this PR does and why we need it:
Bump bundle lib for next release

#### Changes proposed in this pull request
 - re-adding origin copy back
 - rename apb -> bundle package

#### Does this PR depend on another PR (Use this to track when PRs should be merged)
This depends on a release branch being created and master to open up for new development.

### CODE REVIEW

File: pkg/origin/convert.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/openshift/api/authorization/v1"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+)
+
+//ConvertAPIPolicyRulesToRBACPolicyRules -  Convert Policy Rules.
+func ConvertAPIPolicyRulesToRBACPolicyRules(in []v1.PolicyRule) []rbac.PolicyRule {
+	rules := make([]rbac.PolicyRule, 0, len(in))
+	for _, rule := range in {
+		// We need to split this rule into multiple rules for RBAC
+		if isResourceRule(&rule) && isNonResourceRule(&rule) {
```
Comment (maleck13) [inline pkg/origin/convert.go]: Nit: This reads a bit strange but I can understand what it is doing. I had to read the definitions of the functions to get it though

**Comment by maleck13** [comment]
@shawn-hurley seeing an issue on this branch with the watchBundle:
```

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11255d5]

goroutine 1396 [running]:
github.com/openshift/ansible-service-broker/vendor/github.com/automationbroker/bundle-lib/runtime.provider.WatchRunningBundle(0x1a726a0, 0x26c2978, 0x1a88120, 0x26c2978, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/kelly/projects/src/github.com/openshift/ansible-service-broker/vendor/github.com/automationbroker/bundle-lib/runtime/runtime.go:392 +0x65
github.com/openshift/ansible-service-broker/vendor/github.com/automationbroker/bundle-lib/bundle.(*executor).Deprovision.func1(0xc4209b0280, 0xc4204432c0)
	/Users/kelly/projects/src/github.com/openshift/ansible-service-broker/vendor/github.com/automationbroker/bundle-lib/bundle/deprovision.go:66 +0x20e
created by github.com/openshift/ansible-service-broker/vendor/github.com/automationbroker/bundle-lib/bundle.(*executor).Deprovision
/Users/kelly/projects/src/github.com/openshift/ansible-service-broker/vendor/github.com/automationbroker/bundle-lib/bundle/deprovision.go:36 +0x2d4 
```

looks like watchBundle is nil. I think it is an issue with the bundle-lib where the defaultWatchRunningBundle isn't assigned in the provider. Not sure if this should be done via the broker? I would have thought this default would be set in the ```NewRuntime``` function?

**Comment by maleck13** [comment]
Not sure if these are related to my local environment but also seeing the following error during provision:
```
[2018-05-15T10:40:07.276Z] [DEBUG] - service_id: b43a4272a6efcaaa3e0b9616324f1099
[2018-05-15T10:40:07.276Z] [DEBUG] - plan_id: db1cdb40646cd408a924445e2b95a1bf
[2018-05-15T10:40:07.276Z] [DEBUG] - operation:  68c9c8a9-cfb0-4b3f-8ad6-a3676ad57187
[2018-05-15T10:40:07.283Z] [DEBUG] - state: in progress
[2018-05-15T10:40:07.283Z] [DEBUG] - description: action started
172.17.0.4 - - [15/May/2018:10:40:07 +0000] "GET /ansible-service-broker/v2/service_instances/275e125a-6aba-41a1-bda5-f9a2350772b9/last_operation?operation=68c9c8a9-cfb0-4b3f-8ad6-a3676ad57187&plan_id=db1cdb40646cd408a924445e2b95a1bf&service_id=b43a4272a6efcaaa3e0b9616324f1099 HTTP/1.1" 200 64
time="2018-05-15T10:40:07Z" level=info msg="command output: rpc error: code = 13 desc = invalid header field value \"oci runtime error: exec failed: container_linux.go:247: starting container process caused \\\"exec: \\\\\\\"broker-bind-creds\\\\\\\": executable file not found in $PATH\\\"\\n\"\r\n - err: "
time="2018-05-15T10:40:07Z" level=info msg="retry attempt: 13 pod: apb-f692c315-281d-4feb-96e4-ebf7e3849531 in namespace: dh-hello-world-db-apb-prov-xblqw failed to exec into the container"
time="2018-05-15T10:40:12Z" level=info msg="command output:  - err: "
time="2018-05-15T10:40:12Z" level=info msg="retry attempt: 14 pod: apb-f692c315-281d-4feb-96e4-ebf7e3849531 in namespace: dh-hello-world-db-apb-prov-xblqw failed to exec into the container"
time="2018-05-15T10:40:17Z" level=info msg="pod: apb-f692c315-281d-4feb-96e4-ebf7e3849531 in namespace: dh-hello-world-db-apb-prov-xblqw has been completed"
time="2018-05-15T10:40:17Z" level=error msg="apb::provision error occurred - unexpected end of JSON input"
time="2018-05-15T10:40:17Z" level=info msg="Destroying APB sandbox..."
```

**Comment by shawn-hurley** [comment]
@maleck13 any chance you can submit the patch to bundle lib?

**Comment by maleck13** [comment]
@shawn-hurley looks to me like there is an attempt to exec into the container. Not sure why yet digging in

File: pkg/origin/convert.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package origin
+
+import (
+	"github.com/openshift/api/authorization/v1"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+)
+
+//ConvertAPIPolicyRulesToRBACPolicyRules -  Convert Policy Rules.
+func ConvertAPIPolicyRulesToRBACPolicyRules(in []v1.PolicyRule) []rbac.PolicyRule {
```
Comment (eriknelson) [inline pkg/origin/convert.go]: :+1:, happy if we can rip this out of bundle-lib.

**Comment by jmrodri** [comment]
I retested this and things worked fine. Let me know if we should merge.

**Comment by maleck13** [comment]
Something looks to have changed with the state work mount looks like this
```
Mount: 6a86256e-d766-4144-af44-7ebecef16f46-state → /etc/apb/state 
```
but the actual config map is now copied over as
```
bundle-50f5623c-f6bc-4278-adf7-f590ae12b72c
```

investigating

**Comment by maleck13** [comment]
The issue here is the mount is using the configmap name from the broker ns. This is set on the execution context as StateName but actually the configmap in the temporary ns is named after the pod. So I think the following needs to change
https://github.com/automationbroker/bundle-lib/blob/master/runtime/run_bundle.go#L64
to use ```BundleName``` 
I will create a PR

**Comment by jmrodri** [comment]
rebased to pick up the travis changes from @djzager

**Pull request merged**